swiftsimio/writer.py

```diff
@@ -200,6 +200,120 @@ def write_particle_group(self, file_handle: h5py.File, compress: bool):
 
         return
 
+    def write_particle_group_metadata(
+        self, file_handle: h5py.File, dset_attributes: dict
+    ):
+        """
+        Writes the relevant metadata for a particle group
+
+        Parameters
+        ----------
+        file_handle: h5py.File
+            hdf5 output file being written
+        dset_attributes: dict
+            dictionary containg metadata to attach to group
+        """
+        for name, output_handle in getattr(
+            metadata.required_fields, self.particle_name
+        ).items():
+            obj = file_handle[f"/PartType{self.particle_type}/{output_handle}"]
+            for attr_name, attr_value in dset_attributes[output_handle].items():
```

Comment:
Maybe I am wrong, but can we not just do:

```
obj.attrs = dset_attributes[output_handle]
```

Reply:
Unfortunately not. it claims it can't set the attribute that way.

Reply:
:( Boo. Thanks for checking.